sql/core/src/test/resources/sql-tests/inputs/cast_op.sql

```diff
@@ -0,0 +1,148 @@
+-- cast string representing a valid fractional number to integral should truncate the number
+SELECT '1.23' :: int;
+SELECT '1.23' :: long;
+SELECT '-4.56' :: int;
+SELECT '-4.56' :: long;
+
+-- cast string which are not numbers to numeric types
+SELECT 'abc' :: int;
+SELECT 'abc' :: long;
+SELECT 'abc' :: float;
+SELECT 'abc' :: double;
+
+-- cast string representing a very large number to integral should return null
+SELECT '1234567890123' :: int;
+SELECT '12345678901234567890123' :: long;
+
+-- cast empty string to integral should return null
+SELECT '' :: int;
+SELECT '' :: long;
+SELECT '' :: float;
+SELECT '' :: double;
+
+-- cast null to integral should return null
+SELECT NULL :: int;
+SELECT NULL :: long;
+
+-- cast invalid decimal string to numeric types
+SELECT '123.a' :: int;
+SELECT '123.a' :: long;
+SELECT '123.a' :: float;
+SELECT '123.a' :: double;
+
+-- '-2147483648' is the smallest int value
```

Comment:
I don't think we need so many tests tho.

Reply:
Can we just have few tests? A lot of tests are duplicates with `CastingSyntaxSuite` too

Reply:
maybe just add a few end-to-end tests in `cast.sql`? The underlying expression is the same so we probably don't need to test so many different values.

Reply:
I agree, it is better to test more corner cases:
1. Chain of casts: `<expr> :: type1 :: type2`
2. Regular cast + new one: `CAST(<expr> :: type1 AS type2)`
3. Precedence of operators: `map[1] :: type1 [2]`
4. If you support only primary expr, add a negative test for non-primary expr
5. Incorrect taget type: `<expr> :: BINT`, `<expr> :: SELECT`

Reply:
Addressed all the feedback
 - reduced the number of tests
 - consolidated tests in cast.sql
 - added tests suggested by Max